docs/source/extensions.rst

```diff
@@ -3,17 +3,233 @@
 Environment extensions
 ======================
 
-The environment extension is our fuctionality that allows agent to infer latent observations that are
+The environment extension is our functionality that allows agent to infer latent observations that are
 not originally supported by the environment. You can either choose one of our built-in extensions or
-implemet your own with the help of this short guide.
+implement your own with the help of this short guide.
+
+.. _key_concepts:
+
+Key concepts
+------------
+
+The main axis of this module is the :ref:`abstract class <base_ext>` ``BaseExt``, which provides the core
+functionality of extensions. It implements important methods, such as:
+
+* ``get_agent_params`` - this method is responsible for providing default arguments specified by extension
+  while creating new agent,
+* ``transform`` - composes observations passed by the user with values calculated or defined by the extension
+  to create set of parameters required by agent functions. In other words, ``transform`` fills missing
+  values and returns full set of required parameters,
+* ``setup_transformations`` - based on the update and sample observation spaces of the agent, method creates
+  functions that combine observations passed by the user with values calculated by the extension which are
+  used by the ``transform`` method.
+
```

Comment:
A może wogule wywalić ten fragment opisujący te trzy metody albo opisać go tak ogólnikowo? Bo czy użytkownik na pewno potrzebuje wiedzieć co siedzi we wnętrznościach tej biblioteki? nie wiem, no coś bym tu zmienił, ale nie wiem jak to ugryźć...